frontend/lib/App.dart

```diff
@@ -41,9 +40,18 @@ class _AppState extends State<App> {
   Widget build(BuildContext context) {
     final MediaQueryData queryData = MediaQuery.of(context);
 
-    return queryData.size.width > 560
-        ? _buildRailNavigationScaffold()
-        : _buildBottomNavigationScaffold();
+    return FutureBuilder<List<EndpointSummary>>(
+        future: widget.repository.getEndpointSummary(),
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.none ||
+              snapshot.data == null) {
+            return loadingInCenter();
+          } else {
+            return queryData.size.width > 560
+                ? _buildRailNavigationScaffold()
+                : _buildBottomNavigationScaffold();
+          }
+        });
   }
```

Comment:
Czemu tutaj wywołujemy `repository.getEndpointSummary` skoro nigdzie nie jest używane?

Reply:
ładujemy Listę endpointSummary do dalszego użycia w aplikacji

Reply:
Myślę, że lepszym rozwiązaniem byłoby napisać funkcję asynchroniczną, która wywołuje zapytanie, a potem na `Future.then` zapisuje te dane to cash'a. Bo tego w tym momencie nie widać co napisałeś

Reply:
chcesz w future: funkcja() to zrobić tak?

Reply:
dokładnie to jest robione w RestEndpointrepository

Reply:
To nie powinna być rola repozytorium. Repozytorium służy do zarządzania danymi (zapis, odczyt, usuwanie, update) z jednego źródła (API, baza danych, cash lokalny, itd.). A tutaj repozytorium jednocześnie wybiera, z którego źródła ma czerpać dane. Myślę, że sensowniejsze będzie ten wybór wyrzucić gdzieś wyżej, żeby repozytorium samo w sobie nie było za niego odpowiedzialne